### ISSUE TITLE
UnrecognizedClientException error when changing Access & secret key

### ISSUE THREAD

**Issue by eunice98k** [grafana/athena-datasource#246]
After changing the Access & secret key on Athena data source configuration, I am seeing `UnrecognizedClientException` error.
It seems like access key ID and secret access key are cached at a datasource level in athena plugin (v2.2.0): https://github.com/mtanda/grafana-aws-athena-datasource/blob/2.2.0/credentials.go#L24-L223

 Here is full error message:

```
error querying the database: error executing query: UnrecognizedClientException: The security token included in the request is invalid. status code: 400
```
<img width="1169" alt="Screenshot 2023-05-24 at 10 08 20 AM" src="https://github.com/mtanda/grafana-aws-athena-datasource/assets/43913498/db9ec940-e9f3-4413-84db-e1834766ad68">

**Steps to reproduce**

- Create an Athena data source. The authentication provider choose an access & secret key and input the access key ID and secret access key.
- Use Explore to query in Athena.
- Create a new IAM user and obtain an Access Key ID and Secret Access Key.
- Delete the IAM user that used as the data source in step 1.
- Change the Access Key ID and Secret Access Key of data source to those obtained in step 3 and save them.
- Use Explore to query in Athena, then I get the same error as the customer.
- The error continues for while, then resolves without any action.

**Comment by robbierolin** [comment]
Yes the issue exists in Amazon Managed Grafana which uses the plugin from this repo. Version 2.8.0 was being used at the time we saw this issue.

**Comment by iwysiu** [comment]
I was able to reproduce this with Grafana v9.5.3 and Athena v2.8.0. I'll try to take a deeper look at it this week.

**Comment by iwysiu** [comment]
After some further investigating, I confirmed that we're caching the session with the old credentials. When we get the Athena interface for the datasource, it creates it with the datasource id as the session key. This means that even though the credentials have been changed, when the Athena datasource [gets the session for the new credentials](https://github.com/grafana/athena-datasource/blob/5c9f6a3e1b16f7870bfe3165391807e13fad46e0/pkg/athena/datasource.go#L104), the [old one is loaded and returned instead](https://github.com/grafana/grafana-aws-sdk/blob/0273014e4cd566554bf6f835d3037f40786c42c5/pkg/sql/datasource/datasource.go#L190). We should change how this caching works (maybe change the session key) to fix this.

I'll move this to our backlog for now.

**Comment by sarahzinger** [comment]
@iwysiu is there a work around we can recommend to clear the cache for folks running into this problem until we can fix it? If not maybe we should move it to next.

**Comment by KunalEXA** [comment]
Hey @iwysiu 
Just looking to understand if the fix for this issue has been prioritized.

Thanks!

**Comment by iwysiu** [comment]
Hi @KunalEXA ! Yes, its been prioritized. There should be someone working on it soon.

**Comment by idastambuk** [comment]
Hi @eunice98k and @KunalEXA we have released the fix for this with Athena@2.10.2 - does upgrading the plugin solve your issue?
